elf_binary.cc

```diff
@@ -66,23 +66,33 @@ Range ELFBinary::GetRange() const {
     return range;
 }
 
-bool ELFBinary::InTLS(uintptr_t offset) const {
+bool ELFBinary::IsVaddrInTLSData(uintptr_t vaddr) const {
     if (tls_) {
-        return tls_->p_vaddr <= offset && offset < tls_->p_vaddr + tls_->p_memsz;
+        uint64_t a = reinterpret_cast<uint64_t>(vaddr);
```

Comment:
> In addition to it, I rewrite InTLS, InTLSData, and InTLSBSS not to compare a pointer and an integer directly.

Please note `uintptr_t` is an integer type, not a pointer. I believe there were only safe integer comparisons and we do not need any casts here?

I think `reinterpret_cast` is not preferred in general. E.g., https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#prosafety-type-safety-profile

A general principle is "always avoid `reinterpret_cast` when other casts, especially `static_cast` works. I'm not sure my knowledge is up-to-date, but here's a list of good practices I remember:

* `static_cast`: int <=> int, int <=> float
* `static_cast`: void* <=> other pointer types
* `reinterpret_cast`: non-void* pointer types <=> other pointer types
* `reinterpret_cast`: pointer types <=> `intptr_t`/`uintptr_t`

Reply:
Oops, I assumed `uintptr_t` is a pointer. So, we don't need any cast here, as you said.